### PR TITLE
refactor bigdl structure

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?

refactor bigdl structure

### CODE REVIEW

File: make-dist.sh [hunk 1]
```diff
@@ -68,7 +68,7 @@ cp $BASEDIR/scripts/bigdl.sh $BIN_DIR/
 cp $BASEDIR/scripts/classes.lst $BIN_DIR/
 cp $BASEDIR/scripts/img_class.lst $BIN_DIR/
 
-cp $BASEDIR/dl/target/bigdl-$VERSION-jar-with-dependencies.jar $LIB_DIR/
-cp $BASEDIR/dl/target/bigdl-$VERSION-jar-with-dependencies-and-spark.jar $LIB_DIR/
-cp $BASEDIR/dl/target/bigdl-$VERSION-python-api.zip $LIB_DIR/
+cp $BASEDIR//spark/dl/target/bigdl-$VERSION-jar-with-dependencies.jar $LIB_DIR/
```
Comment (jason-dai) [inline make-dist.sh]: //spark/ or /spark/?

File: spark/dl/pom.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <parent>
-        <artifactId>bigdl-parent</artifactId>
+        <artifactId>spark</artifactId>
```
Comment (jason-dai) [inline spark/dl/pom.xml]: can we use a different name than spark?

File: pom.xml [hunk 3]
```diff
@@ -177,8 +177,7 @@
     </properties>
 
     <modules>
-        <module>dl</module>
-        <module>spark-version</module>
+        <module>spark</module>
```
Comment (jason-dai) [inline pom.xml]: can we use a different name than spark?

File: spark/pom.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>bigdl-parent</artifactId>
+        <groupId>com.intel.analytics.bigdl</groupId>
+        <version>0.1.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>spark</artifactId>
```
Comment (jason-dai) [inline spark/pom.xml]: can we use a different name than spark?

File: spark/spark-version/pom.xml [hunk 5]
```diff
@@ -3,7 +3,7 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <parent>
-        <artifactId>bigdl-parent</artifactId>
+        <artifactId>spark</artifactId>
```
Comment (jason-dai) [inline spark/spark-version/pom.xml]: can we use a different name than spark?

**Comment by yiheng** [comment]
LGTM

File: make-dist.sh (hunk 1 above)
Comment (zhangxiaoli73) [inline make-dist.sh]: yeah, should be /spark/.Changed

File: spark/dl/pom.xml (hunk 2 above)
Comment (zhangxiaoli73) [inline spark/dl/pom.xml]: yeah, but I am not sure what the name should be, do you have any idea?

Comment (jason-dai) [inline spark/dl/pom.xml]: why not keep it as bigdl-parent? or simply bigdl?

Comment (zhangxiaoli73) [inline spark/dl/pom.xml]: there are bigdl and spark-version in the file spark, then I add a pom.xml for spark file, and named a module spark. So spark module is bigdl module and spark-version module's parent. Do we need add such spark module?

**Comment by zhangxiaoli73** [comment]
jenkins pass

**Pull request merged**